pairing/bn256/suite_test.go

```diff
@@ -8,6 +8,22 @@ import (
 	"golang.org/x/crypto/bn256"
 )
 
+func TestScalarMarshal(t *testing.T) {
+	suite := NewSuiteBN256()
+	a := suite.G1().Scalar().Pick(random.New())
```

Comment:
This test would be more "convincing" if it made sure that a ended up a value that caused len(a.Bytes()) < a.MarshalSize(). As it is, this will only exercise that code path once per 256 executions on average (i.e. ones where the MSB is zero)

Reply:
Added another test that should trigger that case more regularly.